src/OSPSuite.Core/Domain/Module.cs

```diff
@@ -44,6 +45,15 @@ public class Module : ObjectBase, IEnumerable<IBuildingBlock>
       /// </summary>
       public bool IsPKSimModule { get; set; }
 
+      /// <summary>
+      /// Holds a snapshot of everything required to recreate the module in PK-sim
+      /// </summary>
+      public string Snapshot
+      {
+         set => _snapshot = value;
+         get => !IsPKSimModule ? string.Empty : _snapshot;
```

Comment:
I left this as a hidden snapshot when the module is modified. If the user reverts the change, the snapshot will once again be available, if the project is closed and opened, then the snapshot mapping will be removed by the serialization cycle.

Reply:
How about keeping it, so the user could "reset" the module even it has been converted from PK-Siom to extension?=

Reply:
I think the if is backward. IsPKsimModule ? _snapshot : string.empty

Reply:
well, it's logically correct, but you mean why not remove the `!`

Reply:
> How about keeping it, so the user could "reset" the module even it has been converted from PK-Siom to extension?=

Yeah, you can, as long as you have not re-opened. You cannot undo commands after the project is closed.

Reply:
Oh, I get it. Like even if you have a module converted to extension, you could revert that even if the project had been saved

Reply:
> well, it's logically correct, but you mean why not remove the `!`

yes for readability. It's weird to start with a negation first especailly in a tertiary operator

Reply:
> How about keeping it, so the user could "reset" the module even it has been converted from PK-Siom to extension?=

This would not be enough to set the flag to true. It would have to be a "reload" module kind of action. But yeah keeping it does not hurt.